File: odd_occurrences_in_array/odd_occurrences_in_array.py
# https://app.codility.com/demo/results/trainingRUHE58-CX5/


def solution(A):
    map = {}

    for number in A:
        if (isinstance(map.get(number), int)):
            map[number] = map[number] + 1
        else:
            map[number] = 1

    result = -1

    for number in map:
        if (map[number] % 2 != 0):
            result = number

    return result

File: odd_occurrences_in_array/odd_occurrences_in_array.js
// https://app.codility.com/demo/results/trainingSHE9FS-DF9/

function solution(A) {
  const map = {};

  for (let i = 0; i < A.length; i++) {
    const value = String(A[i]);
    map[value] = (map[value] || 0) + 1;
  }

  const solo = Object.keys(map).filter(key => map[key] % 2 != 0);

  return Number(solo[0]);
}
